pylint/checkers/classes.py

```diff
@@ -908,6 +908,13 @@ def _check_unused_private_functions(self, node: astroid.ClassDef) -> None:
             function_def = cast(astroid.FunctionDef, function_def)
             if not is_attr_private(function_def.name):
                 continue
+            if isinstance(function_def.parent.scope(), astroid.FunctionDef):
```

Comment:
If you move `outer_fn`, you can reuse it here. Would `parent_scope` be a more fitting name?